src/components/program/Program.tsx

```diff
@@ -0,0 +1,100 @@
+'use client';
+
+import type { Session } from "../../types/program.ts";
+import {dayAndTimeFormatWithMonth} from "../../utils/dateformat.ts";
+import {SessionCard} from "./SessionCard.tsx";
+import {useEffect, useState} from "react";
+
+interface Props {
+    timeSlots: string[];
+    sessions: Record<string, Session[]>;
+    filter: string;
+}
+
+function filterSessionTerms(
+    session: Session,
+    liveMode: boolean,
+    now: Date,
+    futureTime: Date,
+): boolean {
+    const startTime = session.startSlot ? new Date(session.startSlot) : undefined;
+    if (startTime === undefined) {
+        return false;
+    }
+
+    const isLive = liveMode && startTime >= now && startTime <= futureTime;
+
+    return (!liveMode || isLive);
+}
+
+export const Program = ({ timeSlots, sessions, filter}: Props) => {
+    const [clicked, setClicked] = useState(false)
+
+    const sessionsById = Object.values(sessions).flat().reduce((acc: Record<string, Session>, session: Session) => {
+        acc[session.id] = session;
+        return acc;
+    }, {});
+
+    function getFavoriteSessions(): Session[] {
+        const favoriteIds = JSON.parse(localStorage.getItem('favorites') || '[]');
+        return favoriteIds.map((id: string) => sessionsById[id]).filter((session: Session | undefined): session is Session => !!session);
+    }
+
+    let favoritesArray: Session[]= getFavoriteSessions();
+
+    useEffect(() => {
+        favoritesArray = getFavoriteSessions();
+    }, [clicked]);
+
+    if (filter == "LIVE"){
+        const now = new Date();
+        now.setMinutes(0, 0, 0);
+        const futureTime = new Date(now);
+        futureTime.setHours(now.getHours() + 2);
+        timeSlots = timeSlots.filter((time) => {
```

Comment:
Det er et anti-pattern å endre props direkte. Du kan heller bruke en ny variabel som du bruker nedover i rendringen. Denne IF-en her kan forøvrig også være en ny funksjon.

Reply:
38a549fa